### PR TITLE
krew added

### PR DESCRIPTION
**Pull request by evalsocket**
## Read then delete

# TL;DR
1. Krew plugin and bot added for publish plugin 

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/lyft/flyte/issues/121

## Follow-up issue
https://github.com/lyft/flyte/issues/121

### CODE REVIEW

**Comment by evalsocket** [comment]
We will merge this PR after our release binary is ready in github release

**Comment by kumare3** [comment]
Looks good to me, but I cannot verify if this works :)

**Comment by evalsocket** [comment]
@kumare3 it will need some work

**Comment by evalsocket** [comment]
Krew success log. I have only one concern, Krew releases bot through an error if it's prerelease because krew doesn't allow anyone to publish their pre-release. 
```bash
➜  flytepropeller git:(feature/krew) docker run -v /Users/evalsocket/hackerspace/golang/src/github.com/lyft/flytepropeller/.krew.yaml:/tmp/template-file.yaml rajatjindal/krew-release-bot:v0.0.38 krew-release-bot template --tag v0.5.12 --template-file /tmp/template-file.yaml
time="2020-12-10T15:13:43Z" level=info msg="getting sha256 for https://github.com/lyft/flytepropeller/releases/download/v0.5.12/kubectl-flyte_v0.5.12_darwin_x86_64.tar.gz"

time="2020-12-10T15:16:22Z" level=info msg="downloaded file /tmp/460437544/1607613223"
time="2020-12-10T15:16:22Z" level=info msg="getting sha256 for https://github.com/lyft/flytepropeller/releases/download/v0.5.12/kubectl-flyte_v0.5.12_linux_x86_64.tar.gz"

time="2020-12-10T15:18:28Z" level=info msg="downloaded file /tmp/967630695/1607613382"
apiVersion: krew.googlecontainertools.github.com/v1alpha2
kind: Plugin
metadata:
  name: flyte
spec:
  version: "v0.5.12"
  homepage: https://github.com/lyft/flyt
  shortDescription: >-
  description: |+2
      kubectl-flyte is an command line tool that can be used as an extension to kubectl.
  platforms:
    - selector:
        matchLabels:
          os: darwin
          arch: amd64
      uri: https://github.com/lyft/flytepropeller/releases/download/v0.5.12/kubectl-flyte_v0.5.12_darwin_x86_64.tar.gz
      sha256: 7ccfe0f2cda4f9ff613191f770d62338aa24ebea923c8748538fd84b11bbc7b3
      files:
        - from: kubectl-flyte
          to: ./flyte
        - from: LICENSE
          to: .
      bin: flyte
    - selector:
        matchLabels:
          os: linux
          arch: amd64
      uri: https://github.com/lyft/flytepropeller/releases/download/v0.5.12/kubectl-flyte_v0.5.12_linux_x86_64.tar.gz
      sha256: c96801b6e2ed208d4e110a34d6a5d6b3a4605fb74ce5b197e57c6f9f04f1ecc3
      files:
        - from: kubectl-flyte
          to: ./flyte
        - from: LICENSE
          to: .
      bin: flyte
```

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -97,6 +97,9 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GORELEASER_TOKEN }}
 
+      - name: Update new version in krew-index
```
Comment (EngHabu) [inline .github/workflows/master.yml:100]: Can you add a condition here to detect if it's a prerelease and not proceed?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -97,6 +97,10 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GORELEASER_TOKEN }}
 
+      - if: ! contains(github.ref, "-rc")
```
Comment (EngHabu) [inline .github/workflows/master.yml:100]: I do not think this is enough... there are all kinds of formats people use -alpha -beta -myrelease are all valid pre-release releases...

Also, this WF doesn't actually create a release, it just creates a Tag... goreleaser then creates a release... which triggers this flow but gets skipped due to the conditions we have in the bump version job.

You might want to add a completely new workflow that only handles "Published" event... then add a condition for prereleases like this: https://github.community/t/release-prerelease-action-triggers/17275/2... will this work? I don't know what krew actually errors on... the format of the semver release tag?

**Comment by evalsocket** [comment]
@EngHabu any update on this PR ?

File: .krew.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: flyte
+spec:
+  version: "{{ .TagName }}"
+  homepage: https://github.com/lyft/flyt
```
Comment (EngHabu) [inline .krew.yaml:7]:
```suggestion
  homepage: https://github.com/lyft/flyte
```

**Review by EngHabu — APPROVED** [review]

**Review by EngHabu — APPROVED** [review]

**Review by anandswaminathan — APPROVED** [review]

**Pull request merged**